airbyte-cdk/python/airbyte_cdk/connector.py

```diff
@@ -74,7 +78,10 @@ def spec(self, logger: logging.Logger) -> ConnectorSpecification:
         if yaml_spec:
             spec_obj = yaml.load(yaml_spec, Loader=yaml.SafeLoader)
         elif json_spec:
-            spec_obj = json.loads(json_spec)
+            try:
+                spec_obj = json.loads(json_spec)
+            except json.JSONDecodeError as error:
+                raise ValueError(f"Could not read json spec file: {error}. Please ensure that it is a valid JSON.")
```

Comment:
I would have like to reuse `read_json_file` but if we want to keep which file we were not able to parse, this would have become tricky since we don't read the file the same way for the spec. I'm open to suggestion as how we could reuse in a clean way